joomla/joomla-cms
# Not allowed to core.manage? Use 403 exception (for remaning components)

Pull Request for Improvement.
### Summary of Changes

This PR is the sequence of https://github.com/joomla/joomla-cms/pull/11593.

Replace all remaining administrator components existing 404 JError for a 403 php exception (`JAccessExceptionNotallowed`) when the user does not have access to "Access Administration Interface" (core.manage).
#### Before

![image](https://cloud.githubusercontent.com/assets/9630530/17372250/f2d6984e-599a-11e6-8016-3aa75863ff66.png)
#### After

![image](https://cloud.githubusercontent.com/assets/9630530/17372276/0d4a5b7a-599b-11e6-9bdf-02beaefa567e.png)
### Testing Instructions
- Use latest staging
- Apply patch
- Create a user and add it to "Manager" group
- Go to to all components and set "Access Administration Interface" (core.manage) to "Denied" for "Manager" group in all remaning components (com_content, com_banners, com_contact, com_media, com_newsfeeds, com_search and com_finder)
- Try to access the following URL, you should see 403 errors now:

```
/administrator/index.php?option=com_banners
/administrator/index.php?option=com_cache
/administrator/index.php?option=com_categories
/administrator/index.php?option=com_categories&extention=com_content
/administrator/index.php?option=com_checkin
/administrator/index.php?option=com_contact
/administrator/index.php?option=com_content
/administrator/index.php?option=com_contenthistory&view=history (this one will give a layout not definied error)
/administrator/index.php?option=com_finder
/administrator/index.php?option=com_installer
/administrator/index.php?option=com_joomlaupdate
/administrator/index.php?option=com_languages
/administrator/index.php?option=com_media
/administrator/index.php?option=com_menus
/administrator/index.php?option=com_messages
/administrator/index.php?option=com_modules
/administrator/index.php?option=com_newsfeeds
/administrator/index.php?option=com_plugins
/administrator/index.php?option=com_redirect
/administrator/index.php?option=com_search
/administrator/index.php?option=com_tags
/administrator/index.php?option=com_templates
/administrator/index.php?option=com_users
```
- Code review.

Note the other admin components (com_admin, com_ajax, com_cpanel, com_postinstall) doesn't use this or already use exceptions.

Didn't touch com_config. This one needs another PR.
### Documentation Changes Required

None.
